### PR TITLE
refactor: #2743 send dflymigrate flow from source

### PR DESCRIPTION
**Pull request by BorysTheDev**

### CODE REVIEW

File: src/server/cluster/cluster_family.cc
Comment (dranikpg) [inline src/server/cluster/cluster_family.cc:785]: I'm not an expert in networking, but in the general case the host you've received the message from isn't neccessary the host to send your request to... Your `home pc -> router -> outside world` with port forwarding as an example 🙂 
I _don't_ know how modern cloud networks look and whether there can be such cases

File: src/server/cluster/cluster_family.cc
Comment (dranikpg) [inline src/server/cluster/cluster_family.cc:795]: please DCHECK it's actually true because with async_dispatch everything will break

Comment (dranikpg) [reply src/server/cluster/cluster_family.cc:795]: and add a comment that we do this to be ignored by the dispatch tracker and maybe a todo that we should add a cleaner approach (some kind of separate technical connection flag - like you suggested)

File: src/server/cluster/cluster_shard_migration.cc
Comment (dranikpg) [inline src/server/cluster/cluster_shard_migration.cc:48]: 🤨

File: src/server/cluster/outgoing_slot_migration.h
Comment (dranikpg) [inline src/server/cluster/outgoing_slot_migration.h:38]: Please add a comment what high level ack means because I see it calls Cancel on all flows

File: src/server/cluster/outgoing_slot_migration.cc
Comment (dranikpg) [inline src/server/cluster/outgoing_slot_migration.cc:105]: Is it always called from the same thread as Ack?

**Review by dranikpg — APPROVED** [review]
Please wait for Adis approval. I'm not aware of the high-level design so I can only complain about code 😄

File: src/server/cluster/cluster_family.cc
Comment (BorysTheDev) [reply src/server/cluster/cluster_family.cc:785]: If the node is behind the NAT we can not create a direct connection at all. In any case, it's a temporary code for further changes. I will remove this connection in the next PR

**Pull request closed without merging**
